src/pe/resource.rs

```diff
@@ -0,0 +1,1681 @@
+            )
+            .field(
+                "product_version_ls",
+                &format_args!("{:#x}", &self.product_version_ls),
+            )
+            .field(
+                "file_flags_mask",
+                &format_args!("{:#x}", &self.file_flags_mask),
+            )
+            .field("file_flags", &format_args!("{:#x}", &self.file_flags))
+            .field("file_os", &format_args!("{:#x}", &self.file_os))
+            .field("file_type", &format_args!("{:#x}", &self.file_type))
+            .field("file_subtype", &format_args!("{:#x}", &self.file_subtype))
+            .field("file_date_ms", &format_args!("{:#x}", &self.file_date_ms))
+            .field("file_date_ls", &format_args!("{:#x}", &self.file_date_ls))
+            .finish()
+    }
+}
+
+/// Language and codepage identifier for U.S. English with Unicode (UTF-16) in Windows SDK version info.
+///
+/// This identifier consists of the following components:
+/// - `04`: Primary language identifier for English.
+/// - `09`: Sub-language identifier for United States.
+/// - `04E4`: Codepage identifier for Unicode (UTF-16).
+///
+/// This value may present in [`ResourceString::key`] without no dedicated value data.
+pub const VERSION_INFO_US_ENGLISH_UNICODE: &str = "040904E4";
+/// A [`ResourceString::key`] of [`VsFixedFileInfo`]
+pub const VS_VERSION_INFO_KEY: &str = "VS_VERSION_INFO";
+
+impl<'a> VsFixedFileInfo {
+    /// Returns `true` if [`Self::signature`] equals to [`VS_FFI_SIGNATURE`], otherwise `false`.
+    pub fn is_valid(&self) -> bool {
+        self.signature == VS_FFI_SIGNATURE
+    }
+
+    /// Reinterprets [`VsFixedFileInfo::file_version_ms`] and [`VsFixedFileInfo::file_version_ls`] into a generic [`VersionField`].
+    pub fn file_version(&self) -> VersionField {
+        VersionField::from_ms_ls(self.file_date_ms, self.file_date_ls)
+    }
+
+    /// Reinterprets [`VsFixedFileInfo::product_version_ms`] and [`VsFixedFileInfo::product_version_ls`] into a generic [`VersionField`].
+    pub fn product_version(&self) -> VersionField {
+        VersionField::from_ms_ls(self.product_version_ms, self.product_version_ls)
+    }
+}
+
+/// Represents a collection of string-based file information in a version resource.
+///
+/// This struct holds various metadata attributes about a file, such as the company name,
+/// file description, version information, and copyright details. Each field is optional and
+/// can be absent if the information is not available.
+#[derive(Copy, Clone)]
+pub struct StringFileInfo<'a> {
+    /// Additional information for diagnostic purposes. Can be of arbitrary length.
+    pub comments: Option<&'a [u8]>,
+    /// The name of the company that produced the file, e.g., "Microsoft Corporation".
+    pub company_name: Option<&'a [u8]>,
+    /// A description of the file suitable for presentation to users, e.g., "Keyboard driver for AT-style keyboards".
+    pub file_description: Option<&'a [u8]>,
```

Comment:
so in general, for these utf16 strings, I would in general either pre-parse them (especially if this doesn't occur a lot) to the utf8 owned String, and keep the fields `pub`; _or_ i would make these fields *not* `pub` and only expose the functions returning the parsed version.

in the case of this struct, since there are so many fields, I would almost prefer to pre-parse it, but maybe only if expect there to e.g,. be only one copy of this struct in the resource data; if it's in a vector or iterator and could have a perf issue due to eager pre-parsing, and maybe no one uses, then we can make it private, and have the function logic like you have below